### PR TITLE
Add setup instructions to readme, and refactor for configurability

### PR DESCRIPTION
**Pull request by peterargue**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,74 @@
...
+
+### Build
+The consensus follower requires the `crypto` modules from `onflow/flow-go`, which must be built locally.
+
+Clone the `flow-go` repo
+```
+git clone https://github.com/onflow/flow-go.git
+```
+
+Follow the instructions in the `flow-go` README up to `make install-tools`.
```
Comment (vishalchangrani) [inline README.md:59]:
```suggestion
Follow the instructions in the `flow-go` [README](https://github.com/onflow/flow-go/blob/master/README.md) up to `make install-tools`.
```

File: go.mod [hunk 2]
```diff
@@ -8,4 +8,4 @@ require (
 )
 
 // crypto needs to be locally installed and compiled
-replace github.com/onflow/flow-go/crypto => ./crypto
+replace github.com/onflow/flow-go/crypto => ./flow-go/crypto
```
Comment (vishalchangrani) [inline go.mod:11]: ❤️

**Review by vishalchangrani — APPROVED** [review]
left a small comment but other than that I think thats about all the documentation we need for the consensus follower.

**Comment by smnzhu** [comment]
@vishalchangrani we should probably rename this repo now eh? e.g onflow/full-observer-node

File: main.go [hunk 3]
```diff
@@ -2,80 +2,114 @@ package main
...
+	accessNodeHostname := getEnv("ACCESS_NODE_HOSTNAME", defaultAccessNodeHostname)
+	accessNodeNetworkingPublicKey := getEnv("ACCESS_NODE_NETWORKING_PUBLIC_KEY", defaultAccessNodeNetworkingPublicKey)
+
+	accessNodeLibp2pPort := defaultAccessNodeLibp2pPort
+	port := getEnv("ACCESS_NODE_LIBP2P_PORT", "")
+	if port != "" {
+		accessNodeLibp2pPort, err = strconv.Atoi(port)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Invalid value for ACCESS_NODE_LIBP2P_PORT %v\n", port)
+			fatalError(err)
```
Comment (smnzhu) [inline main.go:53]:
```suggestion
			fatalError(fmt.Errorf("Invalid value %s for ACCESS_NODE_LIBP2P_PORT: %w", port, err))
```

File: main.go [hunk 4]
```diff
@@ -91,6 +125,20 @@ func seedFixture(n int) []byte {
...
+func getEnv(key, fallback string) string {
+	value := os.Getenv(key)
+	if len(value) > 0 {
+		return value
+	}
+	return fallback
+}
+
+func fatalError(err error) {
+	fmt.Fprintf(os.Stderr, "error: %v\n", err)
```
Comment (smnzhu) [inline main.go:137]:
```suggestion
	fmt.Fprintln(os.Stderr, err)
```

I think the "error: ..." is somewhat redundant, it should be plenty obvious that it is an error :)

**Review by smnzhu — APPROVED** [review]
This is great, thanks!

**Comment by peterargue** [comment]
@smnzhu updated

**Comment by vishalchangrani** [comment]
> @vishalchangrani we should probably rename this repo now eh? e.g onflow/full-observer-node

sure - new repo sounds good. I can make that change once this is merged in.

**Review by huitseeker — COMMENTED** [review]
TYSVM!

**Comment by peterargue** [comment]
> @vishalchangrani we should probably rename this repo now eh? e.g onflow/full-observer-node

maybe `onflow/full-observer-node-example` since this is inkintended as an example implementation only.

**Pull request merged**